packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart

```diff
@@ -642,18 +644,52 @@ void select(
     'value:$value, '
     'nonInteraction:$nonInteraction)',
   );
+  final gtagEvent = _gtagEvent(
+    event_category: gac.selectEvent,
+    event_label: selectedItem,
+    value: value,
+    non_interaction: nonInteraction,
+    send_to: gaDevToolsPropertyId(),
+    screenMetrics:
+        screenMetricsProvider != null ? screenMetricsProvider() : null,
+  );
   GTag.event(
     screenName,
-    gaEventProvider: () => _gtagEvent(
-      event_category: gac.selectEvent,
-      event_label: selectedItem,
-      value: value,
-      non_interaction: nonInteraction,
-      send_to: gaDevToolsPropertyId(),
-      screenMetrics:
-          screenMetricsProvider != null ? screenMetricsProvider() : null,
-    ),
+    gaEventProvider: () => gtagEvent,
   );
+
+  final uaEvent = ua.Event.devtoolsEvent(
+    eventCategory: gtagEvent.event_category!,
+    label: gtagEvent.event_label!,
+    value: gtagEvent.value,
+    userInitiatedInteraction: !gtagEvent.non_interaction,
+    userApp: gtagEvent.user_app,
+    userBuild: gtagEvent.user_build,
+    userPlatform: gtagEvent.user_platform,
+    devtoolsPlatform: gtagEvent.devtools_platform,
+    devtoolsChrome: gtagEvent.devtools_chrome,
+    devtoolsVersion: gtagEvent.devtools_version,
+    ideLaunched: gtagEvent.ide_launched,
+    isExternalBuild: gtagEvent.is_external_build,
+    isEmbedded: gtagEvent.is_embedded,
+    ideLaunchedFeature: gtagEvent.ide_launched_feature,
+    g3Username: gtagEvent.g3_username,
+    uiDurationMicros: gtagEvent.ui_duration_micros,
+    rasterDurationMicros: gtagEvent.raster_duration_micros,
+    shaderCompilationDurationMicros:
+        gtagEvent.shader_compilation_duration_micros,
+    traceEventCount: gtagEvent.trace_event_count,
+    cpuSampleCount: gtagEvent.cpu_sample_count,
+    cpuStackDepth: gtagEvent.cpu_stack_depth,
+    heapDiffObjectsBefore: gtagEvent.heap_diff_objects_before,
+    heapDiffObjectsAfter: gtagEvent.heap_diff_objects_after,
+    heapObjectsTotal: gtagEvent.heap_objects_total,
+    rootSetCount: gtagEvent.root_set_count,
+    rowCount: gtagEvent.row_count,
+    inspectorTreeControllerId: gtagEvent.inspector_tree_controller_id,
```

Comment:
Even though we include all of the possible fields, most of these are nullable and if they are null when passed to `Event.fromJson`, those fields won't get sent to GA4